### PR TITLE
ci(vitest): added ci run

### LINKED ISSUE #144: Vitest CI

### PR DESCRIPTION
**Pull request by lukas-mertens**
Closes #144

Signed-off-by: Lukas Mertens <git@lukas-mertens.de>

---

**Stack**:
- #164
- #161
- #160
- #154
- #153
- #140
- #139
- #138
- #136
- #148 ⬅
- #135
- #147
- #134
- #132
- #129
- #130
- #128
- #127
- #126
- #125
- #146

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+jobs:
+  package:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
```
Comment (nzbr) [inline .github/workflows/unit-tests.yml:26]: `.nvmrc` specifies to use node 20. We could just get the version to use from there to ensure consistency

```suggestion
      - name: Read .nvmrc
        id: nvmrc      
        run: |
          echo "node-version=$(cat .nvmrc)" >> $GITHUB_OUTPUT
      - name: Use Node.js
        uses: actions/setup-node@v3
        with:
          node-version: ${{ steps.nvmrc.outputs.node-version }}
```

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          version: 8
+          run_install: false
+      - name: Get pnpm store directory
+        shell: bash
+        run: |
+          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
+      - uses: actions/cache@v3
+        name: Setup pnpm cache
+        with:
+          path: ${{ env.STORE_PATH }}
```
Comment (nzbr) [inline .github/workflows/unit-tests.yml:39]:
```suggestion
        id: pnpm-store
        run: |
          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_OUTPUT
      - uses: actions/cache@v3
        name: Setup pnpm cache
        with:
          path: ${{ step.pnpm-store.outputs.STORE_PATH }}
```

This doesn't leave garbage in the shell environment (doesn't really matter, but I think it's nicer this way)

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (lukas-mertens) [reply .github/workflows/unit-tests.yml:26]: I like the idea. I will do it everywhere!

**Review by nzbr — APPROVED** [review]

**Comment by lukas-mertens** [comment]
✓ Commit merged in pull request [#164](https://github.com/EVerest/everest-admin-panel/pull/164)

**Pull request closed without merging**